scripts/assign_workflow_kapa_qc.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+from EPPs.common import StepEPP, get_workflow_stage
+from pyclarity_lims.entities import Protocol
+
+
+class AssignWorkflow(StepEPP):
+    _use_load_config = False  # prevent the loading of the config file
+    """
+    This script checks the KAPA Next Step udf and assigns the relevant submitted 
+    sample either Request Repeats or Sequencing Plate Preparation defined by the arguments (or does not assign if udf is
+    KAPA Make Normalised Libraries 
+    """
+    def __init__(self, argv=None):
+        super().__init__(argv)
+        self.sppw = self.cmd_args.sppw
+        self.spps = self.cmd_args.spps
+        self.rrw = self.cmd_args.rrw
+        self.rrs = self.cmd_args.rrs
+        self.qw = self.cmd_args.qpcrw
+        self.qs = self.cmd_args.qpcrs
+
+
+    @staticmethod
+    def add_args(argparser):
+        argparser.add_argument('-sw', '--sppw', type=str, required=True, help='Sequencing plate prepration workflow name')
+        argparser.add_argument('-ss', '--spps',  type=str, required=True, help='Sequencing plate preparation step name')
+        argparser.add_argument('-rw', '--rrw', type=str, required=True, help='Request repeats workflow name')
+        argparser.add_argument('-rs', '--rrs',  type=str, required=True, help='Request repeats step name')
+        argparser.add_argument('-qw', '--qpcrw', type=str, required=True, help='Make and Read QPCR workflow name')
+        argparser.add_argument('-qs', '--qpcrs',  type=str, required=True, help='Make and Read QPCR step name')
```

Comment:
Argparse is not designed to handle single-dash options more than one character long.

Reply:
I have changed the single-dash alternatives to arbitrary single character values but the two character values above were working in testing in the system.